src/renderers/dom/shared/wrappers/__tests__/ReactDOMInput-test.js

```diff
@@ -1028,4 +1039,59 @@ describe('ReactDOMInput', () => {
       'node.setAttribute("checked", "")',
     ]);
   });
+
+  describe('assigning the value attribute on controlled inputs', function() {
+    function getTestInput() {
+      return React.createClass({
+        getInitialState: function() {
+          return {
+            value: this.props.value == null ? '' : this.props.value,
+          };
+        },
+        onChange: function(event) {
+          this.setState({ value: event.target.value });
+        },
+        render: function() {
+          var type = this.props.type;
+          var value = this.state.value;
+
+          return (<input type={type} value={value} onChange={this.onChange} />);
+        },
+      });
+    }
+
+    it('always sets the attribute when values change on text inputs', function() {
+      var Input = getTestInput();
+      var stub = ReactTestUtils.renderIntoDocument(<Input type="text" />);
+      var node = ReactDOM.findDOMNode(stub);
+
+      ReactTestUtils.Simulate.change(node, { target: { value: '2'} });
+
+      expect(node.getAttribute('value')).toBe('2');
+    });
+
+    it('does not set the value attribute on number inputs if focused', () => {
+      var Input = getTestInput();
+      var stub = ReactTestUtils.renderIntoDocument(<Input type="number" value="1" />);
+      var node = ReactDOM.findDOMNode(stub);
+
+      node.focus();
+
+      ReactTestUtils.Simulate.change(node, { target: { value: '2'} });
+
+      expect(node.getAttribute('value')).toBe('1');
+    });
+
+    it('sets the value attribute on number inputs on blur', () => {
+      var Input = getTestInput();
+      var stub = ReactTestUtils.renderIntoDocument(<Input type="number" value="1" />);
+      var node = ReactDOM.findDOMNode(stub);
+
+      ReactTestUtils.Simulate.change(node, { target: { value: '2'} });
+      ReactTestUtils.SimulateNative.blur(node);
+
+      expect(node.getAttribute('value')).toBe('2');
+    });
+  });
+
```

Comment:
Does this sufficiently cover this feature? 

1. Coverage of non-text inputs
2. Coverage of focused number inputs
3. Coverage of focused number inputs that get blurred

Any edge cases I'm not thinking of?